### PR TITLE
[Chore] Format code with cargofmt

### PR DESCRIPTION
**Pull request by lloydmeta**
It was getting annoying not being able to use cargo fmt on this project w/o seeing a ton of other diffs, so I ran 'cargo fmt' on the whole project, using an up-to-date version of https://github.com/rust-lang-nursery/rustfmt

### CODE REVIEW

File: core/src/coproduct.rs [hunk 1]
```diff
@@ -781,33 +795,36 @@ pub trait CoproductEmbedder<Out, Indices> {
...
 
 impl CoproductEmbedder<CNil, HNil> for CNil {
-    fn embed(self) -> CNil
-    { match self {
+    fn embed(self) -> CNil {
+        match self {
         // impossible!
-    }}
+    }
+    }
```
Comment (ExpHP) [inline core/src/coproduct.rs]: Looks like my formatting here confused rustfmt.  I may have to break this habit... (and maybe submit a bug to rustfmt)

File: core/src/coproduct.rs [hunk 2]
```diff
@@ -376,9 +379,11 @@ impl<Head, Tail> Coproduct<Head, Tail> {
     /// # }
     /// ```
     #[inline(always)]
-    pub fn subset<Targets, Indices>(self)
-    -> Result<Targets, <Self as CoproductSubsetter<Targets, Indices>>::Remainder>
-    where Self: CoproductSubsetter<Targets, Indices>,
+    pub fn subset<Targets, Indices>(
+        self,
+    ) -> Result<Targets, <Self as CoproductSubsetter<Targets, Indices>>::Remainder>
```
[the remark this answers is not in the document]
Comment (ExpHP) [reply core/src/coproduct.rs]: Oddly enough, I disagree.  This is actually exactly what I'd write on my own, and I never use rustfmt.

File: core/src/coproduct.rs [hunk 3]
```diff
@@ -730,31 +736,39 @@ pub trait CoproductSubsetter<Targets, Indices>: Sized {
...
 
-impl<Choices, THead, TTail, NHead, NTail, Rem>
-    CoproductSubsetter<Coproduct<THead, TTail>, HCons<NHead, NTail>> for Choices
+impl<
+    Choices,
+    THead,
+    TTail,
+    NHead,
+    NTail,
+    Rem,
```
[the remark this answers is not in the document]
Comment (ExpHP) [reply core/src/coproduct.rs]: I agree, blecch!

File: core/src/coproduct.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (Centril) [reply core/src/coproduct.rs]: Personally, I'd write this as:

```rust
pub fn subset<Targets, Indices>(self) // Arguments on a single line
    // The return arrow on its own line
    -> Result<Targets, <Self as CoproductSubsetter<Targets, Indices>>::Remainder>
// The rest is rustfmt standard:
where
    Self: CoproductSubsetter<Targets, Indices>,
{
     CoproductSubsetter::subset(self)
}
```

or with a type alias:

```rust
type CoSubRem<S, T, I> = <S as CoproductSubsetter<T, I>>::Remainder;
type SubsetResult<S, T, I> = Result<Targets, CoSubRem<Self, Targets, Indices>>;

pub fn subset<Targets, Indices>(self) -> SubsetResult<Self, Targets, Indices>
where Self: CoproductSubsetter<Targets, Indices> {
     CoproductSubsetter::subset(self)
}
```

**Comment by lloydmeta** [comment]
Cool thanks; going to merge this now. To be clear, this is just to provide a standard baseline for automated-formatting; obviously, this will evolve over time as rustfmt improves and do-not-format-blocks get tastefully and sparingly added in here and there :)

File: core/src/coproduct.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (ExpHP) [reply core/src/coproduct.rs]: Meh. Cases like this where the argument list is short while the return type is long are uncommon and so it seems odd to me for that to be the one circumstance in which the `->` starts a new line.

I like type aliases too, but only if we can come up with a consistent story for naming them.  Personally I think the docs look nicer where they can show `Self::Remainder`.

[the remark this answers is not in the document]
Comment (Centril) [reply core/src/coproduct.rs]: Formatting OCDs you know ;) I love the semantics of Rust but I do wish it had Haskell syntax ^,-

**Pull request merged**